idb/postgres/postgres_migrations.go

```diff
@@ -122,6 +136,59 @@ func (db *IndexerDb) markMigrationsAsDone() (err error) {
 	return db.SetMetastate(migrationMetastateKey, migrationStateJSON)
 }
 
+func getMigrationState(db *IndexerDb) (*MigrationState, error) {
+	migrationStateJSON, err := db.GetMetastate(migrationMetastateKey)
+	if err == sql.ErrNoRows {
+		// no previous state, ok
+		return nil, nil
+	} else if err != nil {
+		return nil, fmt.Errorf("%s, get m state err", txidMigrationErrMsg)
+	}
+	var txstate MigrationState
+	err = json.Decode([]byte(migrationStateJSON), &txstate)
+	if err != nil {
+		return nil, fmt.Errorf("%s, migration json err", txidMigrationErrMsg)
+	}
+	return &txstate, nil
+}
+
+// Cached values for processAccount
+var hasRewardsSupport = false
+var lastCheckTs time.Time
+
+// processAccount is a helper to modify accounts based on migration state.
+func (db *IndexerDb) processAccount(account *generated.Account) {
+	if hasRewardsSupport {
+		// skip this whole if/else block
+	} else if db.migration != nil {
```

Comment:
To be sneakily thread-safe-ish without a mutex, grab a local copy of db.migration and check if that local copy is not nil. db.migration could go from not nil to nil between check and .GetStatus()

Reply:
I made an attempt to simplify this logic. For the nil check I modified GetStatus to handle the nil case, I've never used that go feature before but I think it works nicely here.